### PR TITLE
Delete a deleted Users Forms

### PR DESCRIPTION
**Pull request by jotoeri**
As we talked about, i was interested to do this. 😄 And it was just too easy to implement. 🙈 
Basically a done PR here, i'd just still wait for a while until we don't see urgent bugs for v2.2.2 anymore. 😉 

Implements #828

### CODE REVIEW

**Comment by skjnldsv** [comment]
Is this ready for reviews?

**Comment by jotoeri** [comment]
It is, yes.
I just didn't want to merge until we know, we don't need a `v2.2.3`. And i didn't know how to mark that ideally.^^

File: lib/Db/FormMapper.php [hunk 1]
```diff
@@ -107,4 +116,15 @@ public function findAllByOwnerId(string $ownerId): array {
...
+
+	/**
+	 * Delete a Form including connected Questions and Submissions
+	 * @param Form $form The form instance to delete
+	 */
+	public function deleteForm(Form $form): void {
+		// Delete Submissions(incl. Answers), Questions(incl. Options) and Form.
+		$this->submissionMapper->deleteByForm($form->getId());
+		$this->questionMapper->deleteByForm($form->getId());
+		$this->delete($form);
```
Comment (skjnldsv) [inline lib/Db/FormMapper.php:128]: Could there be an error here? If so do we need to catch and log?
Maybe also just log debug by default as well?

Comment (jotoeri) [reply lib/Db/FormMapper.php:128]: Logging is done one level higher on API and BackgroundJob respectively.
Regarding errors, this should be ok, as we only delete, what we first got from db. So as long as there is no kind of 'read-only'-Db that should work. 😉

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
Rebase and squash?

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
php cs lint fails

**Comment by jotoeri** [comment]
That's the php-cs-fixer bug... So wie can either ignore it for now, or i'll rebase on the weekend onto #864 😉

**Comment by jotoeri** [comment]
Rebase is done meanwhile, but please don't merge, until #871 is merged and released. 😞

**Review by skjnldsv — CHANGES_REQUESTED** [review]
Blocking then

**Comment by rullzer** [comment]
So good to go now?

**Comment by jotoeri** [comment]
Yeah, should be good... 👍  I just hope we don't need a 2.2.5 🙈

**Review by skjnldsv — APPROVED** [review]

**Comment by jotoeri** [comment]
Just rebased, so i'd say lets go here. 😉

**Pull request merged**